### PR TITLE
Generate navigations from data yamls

### LINKED ISSUE #1621: Generate navigations
After [this](#1588) gets in production we should generate navigations per locale because the current way of building the navigations is to much error prone. Missing links and all that.

### PR DESCRIPTION
**Pull request by RonaldDijkstra**
Using data files to generate the navigations has some advantages: 

- easier maintainable 
- less code
- no more if this root_locale then that logic in templates so more clearer code
- no difference between the desktop, footer & mobile navigation by accident because 1 source, so less error prone

Todos:

- [x] Main
- [x] Mobile
- [x] Footer
- [x] Make a helper for retrieving navigation item data
- [x] Remove unused stuff from locales

Close #1621

### CODE REVIEW

File: source/partials/header/_nav-dropdowns.erb [hunk 1]
```diff
@@ -7,347 +7,64 @@
 
   <div class="dropdown-container" style="transform: translateX(452px); width: 68px; height: 443px;">
```
Comment (sn3p) [inline source/partials/header/_nav-dropdowns.erb]: Not introduced in this PR, but we prolly should apply the styling in the stylesheets?

File: data/navigations_de.yml [hunk 2]
```diff
@@ -0,0 +1,98 @@
+- title: Produkte
+  type: products-group
+  dropdown: products
+  link: /producten.html
+  links:
+    - title: CAPP Learning Management
+      text: Vielfalt des Lernens für motivierte und qualifizierte Mitarbeiter
+      style: item-capp
```
Comment (sn3p) [inline data/navigations_de.yml]: `class` instead of `style`?

```suggestion
      class: item-capp
```

File: data/navigations_de.yml [hunk 3]
```diff
@@ -0,0 +1,98 @@
+- title: Produkte
+  type: products-group
+  dropdown: products
+  link: /producten.html
```
Comment (sn3p) [inline data/navigations_de.yml]: `url` instead of `link`?

```suggestion
  url: /producten.html
```

File: data/navigations_de.yml [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+  links:
+    - title: CAPP Learning Management
+      text: Vielfalt des Lernens für motivierte und qualifizierte Mitarbeiter
+      style: item-capp
+      link: /capp-lms.html
+      icon: icons/learning-pink.svg
```
Comment (sn3p) [inline data/navigations_de.yml]: Maybe the icon can be applied in the class? e.g. using `background-image`

File: source/partials/header/_nav-mobile.erb [hunk 5]
```diff
@@ -16,248 +21,56 @@
...
-                    <%= image_tag "icons/360-feedback.svg", title: t("navigation.products.threesixty.title"), alt: t("navigation.products.threesixty.title") %>
-                  </div>
-                  <div class="product-link-content">
-                    <%= t("navigation.products.threesixty.title") %>
+                <% elsif item.type == "links-group" %>
+
+                  <div class="mobile-links-list">
+                    <ul>
+                      <% item.links.each do |link| %>
+                      <li>
```
Comment (sn3p) [inline source/partials/header/_nav-mobile.erb]: indent?

File: source/partials/header/_nav-mobile.erb [hunk 6]
```diff
@@ -16,248 +21,56 @@
...
-                    <div class="product-link-content">
-                      <%= t("navigation.products.capp-monitor.title") %>
-                    </div>
-                  <% end %>
-                </li>
-              <% end %>
+                    <ul>
+                      <% item.links.each do |product| %>
+                        <li>
+                            <% nav_link_to(product.link, class: "link-container #{product.style}") do %>
```
Comment (sn3p) [inline source/partials/header/_nav-mobile.erb]: too much indent 👀

File: data/navigations_de.yml (hunk 2 above)
Comment (RonaldDijkstra) [reply data/navigations_de.yml]: Middleman doesn't like 'class' key.

File: data/navigations_de.yml (hunk 4 above)
Comment (RonaldDijkstra) [reply data/navigations_de.yml]: Using inline_svg helper for icons, so I don't know about this.

File: data/navigations_de.yml (hunk 3 above)
Comment (RonaldDijkstra) [reply data/navigations_de.yml]: yep

File: source/partials/header/_nav-mobile.erb (hunk 5 above)
Comment (RonaldDijkstra) [reply source/partials/header/_nav-mobile.erb]: Sure

File: source/partials/header/_nav-mobile.erb (hunk 6 above)
Comment (RonaldDijkstra) [reply source/partials/header/_nav-mobile.erb]: Me so sloppy.

File: data/navigations_de.yml (hunk 2 above)
Comment (RonaldDijkstra) [reply data/navigations_de.yml]: `product[:class]`

File: data/navigations_de.yml (hunk 4 above)
Comment (RonaldDijkstra) [reply data/navigations_de.yml]: For styling icons with a class in the svg.

**Comment by RonaldDijkstra** [comment]
Needs rebase

**Pull request merged**